radix-engine/src/track/track.rs

```diff
@@ -17,50 +17,97 @@ use sbor::rust::mem;
 
 use super::interface::{CanonicalPartition, CanonicalSubstateKey, StoreCommit, StoreCommitInfo};
 
+/// A captured sequence of [`StateUpdate`]s to be applied in particular order.
 #[derive(Debug, Clone, PartialEq, Eq, ScryptoSbor, Default)]
 pub struct StateUpdates {
-    /// A set of partitions that were entirely deleted.
-    /// Note: this batch changes should be applied *before* the [`system_updates`] below (i.e.
-    /// allowing a latter individual substate creation to be applied).
-    pub partition_deletions: IndexSet<(NodeId, PartitionNumber)>,
-
-    /// A set of individual substate updates (indexed by partition and substate key).
-    pub system_updates: SystemUpdates,
+    pub elements: Vec<StateUpdate>,
```

Comment:
_(note from author)_

Naming suggestions welcome here.
_(I'd just like to avoid `result.state_updates.state_updates` :p)_

Reply:
Maybe just `updates`?